src/bundles/communication/MqttController.ts

```diff
@@ -0,0 +1,103 @@
+import { connect, type MqttClient } from 'mqtt/dist/mqtt';
+
+export const STATE_CONNECTED = 'Connected';
+export const STATE_DISCONNECTED = 'Disconnected';
+export const STATE_RECONNECTED = 'Reconnected';
+export const STATE_OFFLINE = 'Offline';
+
+/**
+ * Abstraction of MQTT.
+ *
+ * @param connectionCallback Callback when the connection state changed.
+ * @param messageCallback Callback when a message has been received.
+ */
+export class MqttController {
+  private client: MqttClient | null = null;
+  private connected: boolean = false;
+  private connectionCallback: (status: string) => void;
+  private messageCallback: (topic: string, message: string) => void;
+
+  address: string = '';
+  port: number = 8080;
```

Comment:
MQTT default port should be 1883, unless there's a specific reason we're using 8080, which in that case, should be documented.